sound/soc/codecs/hdac_hda.c

```diff
@@ -518,6 +518,8 @@ static void hdac_hda_codec_remove(struct snd_soc_component *component)
 
 	pm_runtime_disable(&hdev->dev);
 	snd_hdac_ext_bus_link_put(hdev->bus, hlink);
+
+	snd_hdac_regmap_exit(hdev);
```

Comment:
There's already a call to `snd_hdac_regmap_exit()` in `snd_hda_codec_cleanup_for_unbind()`, how do they play together?

Reply:
we currently don't call that function so the regmap_exit() is never called.
The entire flow on remove() is not functional for HDaudio, Maybe this patch should be replaced by a better one, the point is that we don't release what we allocated.

Reply:
Hmm, the regmap_exit() is also missing in error goto label of hdac_hda_codec_probe(). 

@plbossart What do you mean by remove() being non-functional? I do test this and I can successfully unload hdac_hda.

Reply:
yes you can unload but see what happens in terms of memory leaks: https://github.com/thesofproject/linux/issues/2195

Reply:
@plbossart ack, just double-checking there is no other brokenness on top. but yeah, the memleaks need to be fixed as well

Reply:
fixed error handling in hdac_hda_codec_probe(), good suggestion @kv2019i